.ci/component_descriptor

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+import dataclasses
+import os
+import yaml
+
+import pkg_resources
+
+import ci.util
+import gci.componentmodel
+import product.model
+
+component_descriptor_base_path = os.path.abspath(ci.util.check_env('BASE_DEFINITION_PATH'))
+component_descriptor_path = os.path.abspath(ci.util.check_env('COMPONENT_DESCRIPTOR_PATH'))
+
+
+def parse_component_descriptor():
+    component_descriptor_v2 = gci.componentmodel.ComponentDescriptor.from_dict(
+        ci.util.parse_yaml_file(component_descriptor_base_path)
+    )
+
+    return component_descriptor_v2
+
+
+def own_component(component_descriptor):
+    return component_descriptor.component(
+        product.model.Component.create(
+            name=ci.util.check_env('COMPONENT_NAME'),
+            version=ci.util.check_env('COMPONENT_VERSION'),
+        )
+    )
+
+
+COMPONENT_SPEC_PROJECT_NAME = 'gardener-component-model'
+
+
+def component_spec_version():
+    with open(
+        os.path.join(
+            os.path.abspath(ci.util.check_env('MAIN_REPO_DIR')),
+            'requirements.txt',
+        ),
+        'r',
+    ) as requirements_file:
+        parsed_requirements = pkg_resources.parse_requirements(requirements_file)
+        if not (component_spec_requirement := next(
+            (r for r in parsed_requirements if r.project_name == COMPONENT_SPEC_PROJECT_NAME),
+            None,
+        )):
+            raise RuntimeError('Unable to find component-spec in requirements.txt')
+        if len(component_spec_requirement.specs) > 1:
+            raise RuntimeError('No exact version specified for component-spec dependency')
+        component_spec_specs = component_spec_requirement.specs[0]
+
+        return component_spec_specs[1]
+
+
+def add_dependence_on_component(
```

Comment:
dependencY

Reply:
that was intentional. Will change it to `add_component_dependency`